crates/net/network/src/transactions/mod.rs

```diff
@@ -1979,27 +1989,28 @@ mod tests {
         let mut established = listener0.take(2);
         while let Some(ev) = established.next().await {
             match ev {
-                NetworkEvent::SessionEstablished {
+                Ok(NetworkEvent::SessionEstablished {
                     peer_id,
                     remote_addr,
                     client_version,
                     capabilities,
                     messages,
                     status,
                     version,
-                } => transactions.on_network_event(NetworkEvent::SessionEstablished {
+                }) => transactions.on_network_event(Ok(NetworkEvent::SessionEstablished {
                     peer_id,
                     remote_addr,
                     client_version,
                     capabilities,
                     messages,
                     status,
                     version,
-                }),
-                NetworkEvent::PeerAdded(_peer_id) => continue,
-                ev => {
+                })),
+                Ok(NetworkEvent::PeerAdded(_peer_id)) => continue,
+                Ok(ev) => {
                     panic!("unexpected event {ev:?}")
                 }
+                Err(err) => panic!("unexpected error {err:?}"),
```

Comment:
we don't want to panic here, let's log the error only

Reply:
sure done in all the panic occurrences when handling events (both for `Err(err)` and `Ok(ev)`)